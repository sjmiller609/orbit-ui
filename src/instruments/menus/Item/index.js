'use strict'
import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import s from './styles.scss'
import { Link } from 'instruments'

const Item = ({
  to,
  children,
  active,
  className,
  activeClassName,
  ...props
}) => {
  return (
    <Link
      {...props}
      to={to}
      exact={true}
      className={classnames(s.item, active && s.active, className)}
      activeClassName={classnames(s.active, activeClassName)}>
      {children}
    </Link>
  )
}

Item.propTypes = {
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
  className: PropTypes.string,
  activeClassName: PropTypes.string,
  to: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
  active: PropTypes.bool,
}

export default Item
