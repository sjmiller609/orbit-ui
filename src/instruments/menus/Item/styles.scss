@import 'src/styles/mixins.scss';

a.item {
  display: block;
  font-size: 1.4rem;
  color: $darkGray;

  &:hover,
  &:focus {
    color: $lightPurple;
    img {
      opacity: 1;
    }
  }

  img {
    opacity: 0.5;
  }
  // this prevents width from changing on hover
  &::after {
    display: block;
    content: attr(title);
    @include font($f700);
    height: 0;
    overflow: hidden;
    visibility: hidden;
  }
}

a.active {
  @include font($f700);
  color: $lightPurple;
}
