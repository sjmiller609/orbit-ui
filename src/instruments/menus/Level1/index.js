'use strict'
import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'

import { H2, Icon, Link, Dropdown, Item, MenuList } from 'instruments'
import s from './styles.scss'

const Level1 = ({ selected, list, addNew, active, className }) => {
  return (
    <Dropdown
      disable={!active}
      selector={
        <H2 className={classnames(s.menu, active && s.active, className)}>
          <React.Fragment>
            {active ? (
              selected.text
            ) : (
              <Link to={selected.to}>{selected.text}</Link>
            )}
            <Icon icon="arrow" className={s.arrow} />
          </React.Fragment>
        </H2>
      }>
      <MenuList>
        {list.map(li => (
          <Item
            key={li.id}
            to={{
              pathname: '/workspaces/' + li.id,
              state: { from: location.pathname },
            }}
            active={li.id === selected.id}>
            {li.label}
          </Item>
        ))}
        <hr className={s.addNew} />
        <Item key="new" to={addNew.to}>
          {addNew.text}
        </Item>
      </MenuList>
    </Dropdown>
  )
}

Level1.propTypes = {
  selected: PropTypes.object,
  addNew: PropTypes.object,
  list: PropTypes.array,
  active: PropTypes.bool,
  className: PropTypes.string,
}

export default Level1
