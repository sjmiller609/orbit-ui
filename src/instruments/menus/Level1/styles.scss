@import 'src/styles/mixins.scss';

.menu {
  margin: 0 1rem;
  opacity: 0.5;
  transition: opacity 250ms ease-in;

  &.active {
    opacity: 1;

    .arrow {
      transform: rotate(90deg);
    }
  }

  &:hover,
  &:focus {
    opacity: 1;
  }

  .text {
    display: inline-block;
    white-space: nowrap;
    overflow: hidden;
    padding: 0.2rem; //to prevent the Workspace title from getting cut off int he app.
  }

  &.active .text {
    max-width: 50vw;
  }
}

.arrow {
  display: inline-block;
  height: 0.6em;
  margin-left: 1rem;
  margin-top: 0.6rem;
  transition: transform 250ms ease-in;
}

.content {
  width: 80%;
  margin: auto;
}

.addNew {
  &:first-child {
    display: none;
  }

  height: 2px;
  width: 100%;
  background: $lightGray;
  border: none;
}
