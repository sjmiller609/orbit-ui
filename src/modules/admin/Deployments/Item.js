'use strict'
import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import s from './styles.scss'
import { TableRow, Box, P, Mini, H3, ShowDate, Button } from 'instruments'

const Item = ({ deployment, className, handleClick }) => {
  const workspaceId = deployment.workspace.id
  const to =
    '/workspaces/' + workspaceId + '/deployments/' + deployment.releaseName
  const config = to + '/configure'

  const columns = [
    <Box key="0" align="flex-start" className={s.title}>
      <H3>{deployment.label}</H3>
      <P>{deployment.releaseName}</P>
    </Box>,
    <Box key="2" align="flex-start" className={s.log}>
      <P className={s.subTitle}>Deployed</P>
      <Mini>
        <ShowDate date={deployment.createdAt} />
      </Mini>
    </Box>,
    <Box key="3" align="flex-start" className={s.log}>
      <P className={s.subTitle}>Updated</P>
      <Mini>
        <ShowDate date={deployment.updatedAt} />
      </Mini>
    </Box>,
    <Box key="1" align="flex-start" className={s.log}>
      <P className={s.subTitle}>Version</P>
      <P>{deployment.version}</P>
    </Box>,
    <Box key="4" align="flex-start" className={s.upgrade}>
      <Button onClick={() => handleClick(config)} to={null}>
        Config
      </Button>
    </Box>,
  ]

  return (
    <TableRow className={classnames(s.item, className)} columns={columns} />
  )
}

Item.propTypes = {
  deployment: PropTypes.object,
  latestVersion: PropTypes.string,
  className: PropTypes.string,
  handleClick: PropTypes.func,
}

export default Item
