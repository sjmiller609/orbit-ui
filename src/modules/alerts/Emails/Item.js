'use strict'
import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import s from './styles.scss'
import { TableRow, Box, B } from 'instruments'

const Item = ({ email, className, ...props }) => {
  const columns = [
    <Box key="0" align="flex-start" className={s.title}>
      <B>{email}</B>
    </Box>,
  ]

  return (
    <TableRow
      {...props}
      className={classnames(s.item, className)}
      columns={columns}
    />
  )
}

Item.propTypes = {
  email: PropTypes.string,
  className: PropTypes.string,
}

export default Item
