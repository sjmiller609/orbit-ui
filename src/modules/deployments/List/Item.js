'use strict'
import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import s from './styles.scss'
import { TableRow } from '../../../instruments'

const Item = ({ deployment, className }) => {
  const columns = [
    <p key="1">{deployment.title}</p>,
    <p key="2">{deployment.release_name}</p>,
    <p key="3">logs</p>,
  ]

  const to = '/deployments/' + deployment.release_name

  return (
    <TableRow
      className={classnames(s.item, className)}
      columns={columns}
      to={to}
    />
  )
}

Item.propTypes = {
  deployment: PropTypes.object,
  className: PropTypes.string,
}

export default Item
