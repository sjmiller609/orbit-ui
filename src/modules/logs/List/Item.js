'use strict'
import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import s from './styles.scss'
import { TableRow, Box, P, ShowDate } from 'instruments'

const Item = ({ log, className }) => {
  const columns = [
    <Box key="0" align="flex-start" className={s.log}>
      <P>
        > <ShowDate date={log.date} /> {log.log}
      </P>
    </Box>,
  ]

  return (
    <TableRow className={classnames(s.item, className)} columns={columns} />
  )
}

Item.propTypes = {
  log: PropTypes.object,
  className: PropTypes.string,
}

export default Item
