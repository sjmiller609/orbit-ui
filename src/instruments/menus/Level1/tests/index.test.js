import React from 'react'
import { shallow } from 'enzyme'
import Item from '../../Item'
import Level1 from '../index'

const props = {
  selected: {
    to: '#',
    id: '123',
    text: 'Workspace Label',
  },
  list: [],
  addNew: {
    to: '/workspaces/new',
    text: 'New Workspace',
  },
  active: true,
  className: 'testClassString',
}

describe('Level1', () => {
  it('should be defined', () => {
    expect(Level1).toBeDefined()
    expect(Item).toBeDefined()
  })

  it('should render correctly', () => {
    const wrapper = shallow(<Level1 {...props} />)
    expect(wrapper).toMatchSnapshot()
  })
})
