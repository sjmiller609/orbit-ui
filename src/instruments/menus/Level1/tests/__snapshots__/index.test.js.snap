// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Level1 should render correctly 1`] = `
<Dropdown
  disable={false}
  selector={
    <H2
      className="menu active testClassString"
    >
      <Row>
        <span
          className="text"
        >
          Workspace Label
        </span>
        <Icon
          className="arrow"
          icon="arrow"
        />
      </Row>
    </H2>
  }
>
  <MenuList>
    <hr
      className="addNew"
    />
    <Item
      exact={true}
      key="new"
      to="/workspaces/new"
    >
      New Workspace
    </Item>
  </MenuList>
</Dropdown>
`;
